manager/src/main/java/io/gingersnapproject/metrics/CacheAccessRecord.java

```diff
@@ -0,0 +1,22 @@
+package io.gingersnapproject.metrics;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiConsumer;
+
+/**
+ * Records cache access latency.
+ * <p>
+ * This interface extends {@link BiConsumer} allowing to use with {@link CompletionStage#whenComplete(BiConsumer)}
+ * methods.
+ * <p>
+ * Exceptions are not an expected outcome and they are recorded as errors. A {@code null} value means a cache miss.
+ *
+ * @since 0.1
```

Comment:
I think we need to avoid `@since` tags given that we're planning a continuous stream of releases instead of following semantic versioning. The git commit/tag history will be the source of truth.